lib/modules/reducer.js

```diff
@@ -22,7 +22,8 @@ export default (state=DEFAULT, action={}) => {
     }
 
     case actions.SET_PAGE: {
-      const referrerFromHistory = !isEmpty(state.currentPage)
+      const historyLen = state.history.length;
+      const referrerFromHistory = !isEmpty(state.currentPage) && historyLen > 1
```

Comment:
On the plus side, having this code here vs in 2X is nice because this is right where code will get changed to fix the history weirdness making it much less likely to get forgotten.